packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -1811,127 +1847,107 @@ class _TimePickerDialogState extends State<_TimePickerDialog> {
-                    width: _kTimePickerWidthLandscape,
-                    height: timePickerHeightLandscape,
-                    child: Column(
-                      children: <Widget>[
-                        Expanded(
-                          child: Row(
-                            children: <Widget>[
-                              header,
-                              Expanded(child: dial),
-                            ],
-                          ),
-                        ),
-                        actions,
-                      ],
-                    ),
-                  );
-              }
-              return null;
-            }
+        final Widget header = _TimePickerHeader(
+          selectedTime: _selectedTime,
+          mode: _mode,
+          orientation: orientation,
+          onModeChanged: _handleModeChanged,
+          onChanged: _handleTimeChanged,
+          use24HourDials: use24HourDials,
+          helpText: widget.helpText,
         );
+
+        switch (orientation) {
+          case Orientation.portrait:
+            picker = Column(
+              mainAxisSize: MainAxisSize.min,
+              crossAxisAlignment: CrossAxisAlignment.stretch,
+              children: <Widget>[
+                header,
+                Expanded(
+                  child: Column(
+                    mainAxisSize: MainAxisSize.min,
+                    children: <Widget>[
+                      // Dial grows and shrinks with the available space.
+                      Expanded(child: dial),
+                      actions,
+                    ],
+                  ),
+                ),
+              ],
+            );
+            break;
+          case Orientation.landscape:
+            picker = Column(
+              children: <Widget>[
+                Expanded(
+                  child: Row(
+                    children: <Widget>[
+                      header,
+                      Expanded(child: dial),
+                    ],
+                  ),
+                ),
+                actions,
+              ],
+            );
+            break;
+        }
         break;
       case TimePickerEntryMode.input:
         picker = Form(
           key: _formKey,
           autovalidate: _autoValidate,
-          child: Container(
-            constraints: const BoxConstraints(maxWidth: _kTimePickerWidthPortrait),
-            child: SingleChildScrollView(
-              child: Column(
-                mainAxisSize: MainAxisSize.min,
-                children: <Widget>[
-                  _TimePickerInput(
-                    initialSelectedTime: _selectedTime,
-                    helpText: widget.helpText,
-                    onChanged: _handleTimeChanged,
-                  ),
-                  actions,
-                ],
-              ),
+          child: SingleChildScrollView(
+            child: Column(
+              mainAxisSize: MainAxisSize.min,
+              children: <Widget>[
```

Comment:
Widget is not needed

Reply:
Generally yes, but flutter/flutter style guide does keep them